### ISSUE TITLE
#<DRb::DRbUnknown:0x007fdfb22d2898> instead of summary

### ISSUE THREAD

**Issue by victornamuso** [sporkrb/spork-testunit#33]
Upon moving to 0.0.7 from 0.0.6, I now get #DRb::DRbUnknown:0x007fdfb22d2898 instead of the summary line (x tests, x assertions, x failures, etc...). Solely rolling back to 0.0.6 in my gemfile fixes the issue. The tests appear to be running because I do get visual feedback of the failure/error if one occurs.

**Comment by nixme** [comment]
Hmm... are you by chance using Test::Unit on MRI 1.9.2?

**Comment by victornamuso** [comment]
Yes

**Comment by nixme** [comment]
Hrm... I'm wondering whether it's better to split up this gem into `spork-minitest` and `spork-testunit` since they both have completely different code paths at this point. We'd also stop breaking people when changes are made like this.

Anyone have a better idea?

**Comment by abelmartin** [comment]
I just came across this.  I'm using MRI 1.8.7, Rails 3.0.10, Test::Unit.  Is there another gem that I should be using for Test::Unit to interpret the <DRb> object?

**Comment by aspiers** [comment]
I have the same problem with 0.0.7, MRI 1.9.3, and Rails 3.1.3.  Downgrading to 0.0.6 fixes it :-(

**Comment by aspiers** [comment]
@nixme: yes, sounds like a good idea to split off `spork-minitest` into a separate gem.

**Comment by aspiers** [comment]
I looked into this further.  The problem is at the top of `testdrb` where it first tries to `require 'minitest/unit'`, and then reverts to `test/unit` on `LoadError`.  This is not a valid approach, because the presence of minitest does not guarantee that you are using it.  I am on MRI 1.9.3 + Rails 3.2.1, and minitest is installed because several gems including `rake` and `rdoc` depend on it - so in fact minitest is pretty much guaranteed to be installed in any modern Rails environment!  However, I am using the `guard-test` gem to launch Spork, so my DRb server is set up to run with `test/unit` not `minitest`.  (This was a deliberate choice because last time I looked, `guard-minitest` was riddled with issues.)

So I strongly recommend that the minitest support should be syphoned off into `spork-minitest` to avoid further calamity.

**Comment by gabeodess** [comment]
I too would like this fixed!

**Comment by djmaze** [comment]
Removed minitest support in my own fork. You might like to use it in the meantime: https://github.com/djmaze/spork-testunit

**Comment by martinos** [comment]
Thanks @djmaze, you version works well with my functional and unit test. However my integration tests are running but I can't see my tests outputs. I will investigate on it.
